data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/AbstractIndexManager.java

```diff
@@ -59,13 +60,13 @@ public abstract class AbstractIndexManager implements IndexManager {
 
     //For matching a string that begins with a "%{" and ends with a "}".
     //For a string like "data-prepper-%{yyyy-MM-dd}", "%{yyyy-MM-dd}" is matched.
-    private static final String TIME_PATTERN_REGULAR_EXPRESSION = "%\\{.*?\\}";
+    static final String TIME_PATTERN_REGULAR_EXPRESSION = "%\\{.*?\\}";
```

Comment:
Not related to your changes, but should be compiling this pattern up front, rather than compiling it every time we match or replace?

Reply:
Makes sense. Updated using regex Pattern upfront.